### ISSUE TITLE
Caller location isn't accurate for `alias_method_chain` deprecation message

### ISSUE THREAD

**Issue by jeroenvisser101** [rails/rails#24858]
### Steps to reproduce
``` bash
rails new reproduce-app
# Add `gem 'meta-tags'` to Gemfile
bundle install
rails s
```

### Expected behavior
It should tell the call came from the gem.

### Actual behavior
It tells it came from `core_ext/module/aliasing.rb`

### System configuration
**Rails version**: 2.3.1

**Ruby version**: 5.0.0.beta4

### Possible fix
``` diff
diff --git a/activesupport/lib/active_support/core_ext/module/aliasing.rb b/activesupport/lib/active_support/core_ext/module/aliasing.rb
index b6934b9..772fb29 100644
--- a/activesupport/lib/active_support/core_ext/module/aliasing.rb
+++ b/activesupport/lib/active_support/core_ext/module/aliasing.rb
@@ -24,7 +24,7 @@ class Module
   #
   # so you can safely chain foo, foo?, foo! and/or foo= with the same feature.
   def alias_method_chain(target, feature)
-    ActiveSupport::Deprecation.warn("alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super.")
+    ActiveSupport::Deprecation.warn("alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super.", caller_locations(1))

     # Strip out punctuation on predicates, bang or writer methods since
     # e.g. target?_without_feature is not a valid method name.
```

**Comment by maclover7** [comment]
Can you provide more specific information about how to reproduce the bug you are seeing? I think this might be a third party issue with `meta-tags`.

**Comment by jeroenvisser101** [comment]
@maclover7 It happens with other gems (`quiet-assets`) too, it's the deprecation warning for Rails 5's deprecation of the `alias_method_chain` method.

The deprecation message should say this: (note the location)

```
DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from block in <module:ControllerHelper> at /usr/local/opt/rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/meta-tags-2.1.0/lib/meta_tags/controller_helper.rb:16)
```

but instead it says:
```
DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from alias_method_chain at /usr/local/opt/rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/activesupport-5.0.0.beta4/lib/active_support/core_ext/module/aliasing.rb:27)
```

**Comment by maclover7** [comment]
👍 change makes sense to me. Please send in a PR if you would like to. :)

**Comment by sblackstone** [comment]
+1, I've been getting the deprecation warnings for alias_method_chain but it only tells me that its occuring in config/application.rb when it does `Bundler.require(*Rails.groups)`

**Comment by jeroenvisser101** [comment]
@sblackstone could you check if #24861 fixes that? That'd be great! :)

**Comment by sblackstone** [comment]
@jeroenvisser101 Confirmed working...

**Comment by jeroenvisser101** [comment]
@sblackstone thanks a lot :)

**Comment by jeroenvisser101** [comment]
@maclover7 any idea how to proceed? Should I add tests? (and if so, how would I do that exactly?)

**Comment by dcosson** [comment]
Should this be fixed in Rails 5.0.1?

I just updated from 4.x to 5.0.1 and am getting this error with the wrong caller location, it says config/application.rb at the `Bundler.require(*Rails.groups)` line.

**Comment by rafaelfranca** [comment]
No, it will be fixed in 5.0.2

**Comment by matthewd** [comment]
This fix was in 5.0.1. 

It pointing at `Bundler.require` is different from the original problem described in this issue (where it was blaming a line inside rails itself). Please do open a new issue for that.

**Comment by rafaelfranca** [comment]
Yeah, it is in 5.0.1
